### PR TITLE
Optimized emscripten logo in generated html

### PR DESCRIPTION
**Pull request by oliversalzburg**
When I was first playing around with emscripten I saw that the default built output was very large. Even though my `.c` file was a couple of bytes, the resulting `.html` file was 101KB. When I investigated the HTML file, I saw that 95KB were taken up by a huge embedded SVG logo.

## Original
image
93.5KB SVG ([source](https://github.com/emscripten-core/emscripten/blob/main/media/powered_by_logo_mini.svg))

## Optimized
image
12KB PNG

Instead of inline XML, this version embed the PNG as a base64 string. Additionally, the logo is cropped and pre-scaled to the desired size. The logo now also uses the `#emscripten_logo` CSS again and applies padding through that path. I feel like this new approach also nests the logo on a slightly better vertical midpoint on the top bar.

### CODE REVIEW

**Comment by sbc100** [comment]
Not a bad idea! 

Having said that we don't really care about the size of the shell.html.. that output file is really just designed as a starting point the useful during development.  We don't expect anyone to actually ship that output or use it in production.   In other works we care a lot about the size of the generated JS and Wasm files but not so much about this file.

@kripken @juj, WDYT?

File: src/shell.css [hunk 1]
```diff
@@ -13,6 +13,8 @@ canvas.emscripten { border: 0px none; background-color: black; }
 #emscripten_logo {
   display: inline-block;
   margin: 0;
+  padding: 6px;
+  width: 265px;
```
Comment (sbc100) [inline src/shell.css:17]: Does the sizer of the image not already determine the size here?  i.e. is this duplicating the information already encoded in the image?

**Comment by sbc100** [comment]
Did the logo used to use the `#emscripten_logo` CSS in the past?  Did I somehow break that in #20661 maybe?

File: src/shell.css (hunk 1 above)
Comment (oliversalzburg) [reply src/shell.css:17]: No. When I resized the image to 265 pixels width, the drop-shadow of the "powered by" text didn't come out very well. So I left the image larger and scale it down via CSS to get a better result. I didn't want to stray away too much from the original look.

**Comment by oliversalzburg** [comment]
@sbc100 I understand that this template serves only as an example and I'm not concerned about download size with this change. I am freshly coming as a user to this project, and having 1000 lines of SVG in the first example output I'm generating is somewhat irritating.

With this change the logo is put on a single line, which makes it much easier to focus on the code at hand and understand how emscripten works.

btw, `emscripten_logo` is also unused in `latest`.

**Comment by sbc100** [comment]
> @sbc100 I understand that this template serves only as an example and I'm not concerned about download size with this change. I am freshly coming as a user to this project, and having 1000 lines of SVG in the first example output I'm generating is somewhat irritating.

I agree. 

I the the base64 imagine is also irritating... but I understand the logic that its less irritating because its fewer lines?  BTW, even though the base64 is only single logical line I'm guessing it will still take up a lot of vertical space in in an editor like mine (and I think most editor) that wrap long lines? 

Regardless I understand the desire to fix this.

What about an alternative idea: we could simply link to permalink of the logo online?  e.g. http://emscripten.something/logo.png? 

> 
> With this change the logo is put on a single line, which makes it much easier to focus on the code at hand and understand how emscripten works.
> 
> btw, `emscripten_logo` is also unused in `latest`.

I see.   Do we know it was ever used in the past, and for what purpose?

**Comment by oliversalzburg** [comment]
I haven't dug into the history to see where the CSS rule was last used :(

I also agree that linking to the image, instead of inlining it, would be even better. I assumed this is not desirable, because the build fragments should be self-contained, and pulling the resource from an emscripten-operated host might even incur bandwidth costs, if a lot of people are running a shell. That's why I tried to stay close to the current approach.

**Comment by kripken** [comment]
This PR makes sense to me. Even if we don't expect people to ship using the default HTML, it's nice for it to be optimized just so it doesn't look weirdly unoptimized. And probably some people do ship with it.

Separately, good question about just linking to the logo online. This URL could work in theory: https://emscripten.org/_static/Emscripten_logo_full.png But, yeah, it might be better to keep the output self-contained as it is now.

File: emcc.py [hunk 2]
```diff
@@ -4225,7 +4225,9 @@ def generate_traditional_runtime_html(target, options, js_target, target_basenam
 
   shell = do_replace(shell, '{{{ SCRIPT }}}', script.replacement())
   shell = shell.replace('{{{ SHELL_CSS }}}', utils.read_file(utils.path_from_root('src/shell.css')))
-  shell = shell.replace('{{{ SHELL_LOGO }}}', utils.read_file(utils.path_from_root('media/powered_by_logo_mini.svg')))
+  shellLogo = utils.read_binary(utils.path_from_root('media/powered_by_logo_shell.png'))
+  shellLogo64 = base64_encode(shellLogo)
```
Comment (sbc100) [inline emcc.py:4229]: Can you call these something like `shell_logo` and `shell_logo_b64`?

**Review by sbc100 — APPROVED** [review]
lgtm % nit

File: emcc.py (hunk 2 above)
Comment (oliversalzburg) [reply emcc.py]: Good call. Changed.

**Pull request merged**